components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -961,6 +961,7 @@ void RewardsServiceImpl::Shutdown() {
     }
   }
 
+  bat_ledger_client_receiver_.reset();
```

Comment:
@aseren @tmancey 
We need this, otherwise test gets DCHECK.
https://bravesoftware.slack.com/archives/CNH9Q6REV/p1659208021378619
```
15188:259:0126/212256.896286:FATAL:bat_ledger.mojom.cc(38551)] Check failed: !connected. BatLedgerClient::LoadURLCallback was destroyed without first either being run or its corresponding binding being closed. It is an error to drop response callbacks which still correspond to an open interface pipe.
0   libbase.dylib                       0x000000010b461b48 base::debug::CollectStackTrace(void**, unsigned long) + 28
1   libbase.dylib                       0x000000010b3598c8 base::debug::StackTrace::StackTrace() + 24
2   libbase.dylib                       0x000000010b377fec logging::LogMessage::~LogMessage() + 156
3   libbase.dylib                       0x000000010b378dc0 logging::LogMessage::~LogMessage() + 12
4   libbase.dylib                       0x000000010b348e68 logging::CheckError::~CheckError() + 36
5   brave_browser_tests                 0x0000000101d3bff0 bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder::OnIsConnectedComplete(bool) + 120
6   libbindings.dylib                   0x000000010a6f0624 mojo::(anonymous namespace)::DetermineIfEndpointIsConnected(base::WeakPtr<mojo::InterfaceEndpointClient> const&, base::OnceCallback<void (bool)>) + 136
7   libbindings.dylib                   0x000000010a6f04cc mojo::(anonymous namespace)::ResponderThunk::IsConnectedAsync(base::OnceCallback<void (bool)>) + 76
8   brave_browser_tests                 0x0000000101d3bf54 std::Cr::default_delete<bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder>::operator()[abi:v160000](bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder*) const + 100
9   brave_browser_tests                 0x0000000101d3bed8 base::internal::BindState<void (bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder::*)(mojo::StructPtr<ledger::mojom::UrlResponse>), std::Cr::unique_ptr<bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder, std::Cr::default_delete<bat_ledger::mojom::BatLedgerClient_LoadURL_ProxyToResponder>>>::Destroy(base::internal::BindStateBase const*) + 36
10  libbase.dylib                       0x000000010b36ed4c scoped_refptr<base::internal::BindStateBase>::~scoped_refptr() + 76
11  brave_browser_tests                 0x0000000100c68398 _ZN4base8internal9BindStateIZN35SearchResultAdTest_AdsDisabled_Test19RunTestOnMainThreadEvE3$_0JNS_17RepeatingCallbackIFvvEEEEE7DestroyEPKNS0_13BindStateBaseE + 28
12  libbase.dylib                       0x000000010b36ed4c scoped_refptr<base::internal::BindStateBase>::~scoped_refptr() + 76
13  brave_browser_tests                 0x0000000100fbe404 base::internal::BindState<void (extensions::(anonymous namespace)::DownloadFileIconExtractorImpl::*)(float, base::OnceCallback<void (std::Cr::basic_string<char, std::Cr::char_traits<char>, std::Cr::allocator<char>> const&)>, gfx::Image), base::internal::UnretainedWrapper<extensions::(anonymous namespace)::DownloadFileIconExtractorImpl, base::RawPtrBanDanglingIfSupported>, float, base::OnceCallback<void (std::Cr::basic_string<char, std::Cr::char_traits<char>, std::Cr::allocator<char>> const&)>>::Destroy(base::internal::BindStateBase const*) + 28
14  libbase.dylib                       0x000000010b36ed4c scoped_refptr<base::internal::BindStateBase>::~scoped_refptr() + 76
15  libnetwork_cpp.dylib                0x000000010f205058 network::(anonymous namespace)::SaveToStringBodyHandler::~SaveToStringBodyHandler() + 108
16  libnetwork_cpp.dylib                0x000000010f205098 network::(anonymous namespace)::SaveToStringBodyHandler::~SaveToStringBodyHandler() + 12
17  libnetwork_cpp.dylib                0x000000010f2035f4 network::(anonymous namespace)::SimpleURLLoaderImpl::~SimpleURLLoaderImpl() + 248
18  libnetwork_cpp.dylib                0x000000010f203668 network::(anonymous namespace)::SimpleURLLoaderImpl::~SimpleURLLoaderImpl() + 12
19  brave_browser_tests                 0x00000001020f50ec brave_rewards::RewardsServiceImpl::Shutdown() + 536
```

Reply:
@atuchin-m thanks, this is part of Rewards. Cc'ing @szilardszaloki @zenparsing.

Reply:
Seems OK to reset the receiver on shutdown. @szilardszaloki what do you think?

Reply:
From our previous investigation with @bridiver, this won't fix the above `DCHECK` in all cases.
I think we should first remove all `std::bind<>()`s/`std::function<>`s, extra proxy layers, etc., use the most modern way in Mojom to set up/tear down these connections, and see if those `DCHECK`s still trigger. At least this is what we settled for with Brian Johnson when doing the previous investigation.

Out of curiosity: why are we running performance tests in non-release builds?

Reply:
We don't run performance tests on non-release builds.

I've added a browser test to verify the switch. That test fails without this change (in CI Static build with dchecks).
In this particular case the reason is clear (see the stack trace) and reseting fixes it.

Reply:
@atuchin-m Okay, cool. It definitely won't hurt (the Ads team already does it). 👍🏻